ols/app/endpoints/ols.py

```diff
@@ -53,15 +54,18 @@ def conversation_request(llm_request: LLMRequest) -> LLMRequest:
         question_validator = QuestionValidator(
             provider=llm_request.provider, model=llm_request.model
         )
+        metrics.llm_calls_total.inc()
         validation_result = question_validator.validate_question(
             conversation_id, llm_request.query
         )
     except LLMConfigurationError as e:
+        metrics.llm_calls_validation_errors_total.inc()
         raise HTTPException(
             status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
             detail={"response": f"Unable to process this request because '{e}'"},
         )
     except Exception as validation_error:
+        metrics.llm_calls_failures_total.inc()
```

Comment:
Seems like these two metric names are reversed?  (validation errors and call failures)

or they are both validation failures and we need a better name for both of them that distinguishes them

(i realize they are just examples and may not even stick around, but they might, or people might copy them, so lets get the foundation right)

Reply:
What I understand is the exception LLMConfigurationError is raised when validating LLM configuration, and the most probable way to raise the generic Exception is a unsuccessful query to OpenAI/BAM API. So they are named as validation_error and call_failures. 

(unrelated to this PR) It may be better to validate the LLM configuration just after loading the configuration file. We can use a Gauge to show whether the configuration is valid, for example `configuration_valid(component="llm") 1` .

Reply:
> What I understand is the exception LLMConfigurationError is raised when validating LLM configuration, and the most probable way to raise the generic Exception is a unsuccessful query to OpenAI/BAM API. So they are named as validation_error and call_failures.

Failure to call the backend should have a clear error type that is distinct from any other failure type

I think this is fundamentally less about the metrics you defined and more about the fact that our error handling+reporting still has some maturing to do.  At a high level there are 3 categories of failure:

1) an external service we rely on failed (BAM/Watson/OpenAI is down/unavailable/timedout or otherwise errored for a reason that is out of our control)
2) An internal error occurred (we got an unexpected exception trying to process the request.  e.g. we failed to parse the response from the llm, we ran out of memory, something else surprising and bad happened)
3) A user input validation error occurred: the user's request was not valid (e.g. api request object was badly formatted, or the api request asked for a model/provider that doesn't exist, or the credentials used to talk to the llm are invalid/rejected)

>  It may be better to validate the LLM configuration just after loading the configuration file.

i think we are doing that validation during load already(@syedriko can confirm).  I agree that we should not be getting runtime errors in the request flow due to an invalid ols config yaml.  However individual api requests can specify a model+provider so it's possible that the api request specifies a model or provider that does not exist in the configuration yaml.